test/reputation-system/reputation-mining-client/client-auto-functionality.js

```diff
@@ -164,6 +165,102 @@ process.env.SOLIDITY_COVERAGE
+                expect(lastSubmitter).to.equal(minerAddress);
+
+                // Validate the root hash matches what the miner submitted
+                const submission = await repCycle.getReputationHashSubmission(minerAddress);
+                expect(submission.proposedNewRootHash).to.equal(rootHash);
+
+                event.removeListener();
+                resolve();
+              } else {
+                await mineBlock();
+              }
+            });
+
+            // After 30s, we throw a timeout error
+            setTimeout(() => {
+              reject(new Error("ERROR: timeout while waiting for 12 hash submissions"));
+            }, 30000);
+          });
+
+          // Forward through most of the cycle duration and wait for the client to submit all 12 allowed entries
+          await forwardTime(MINING_CYCLE_DURATION * 0.9, this);
+          await receive12Submissions;
+
+          const colonyNetworkEthers = await reputationMinerClient._miner.colonyNetwork;
+          const miningCycleComplete = new Promise(function(resolve, reject) {
+            colonyNetworkEthers.on("ReputationMiningCycleComplete", async (_hash, _nNodes, event) => {
+              const newHash = await colonyNetwork.getReputationRootHash();
+              expect(newHash).to.not.equal(oldHash, "The old and new hashes are the same");
+              expect(newHash).to.equal(rootHash, "The network root hash doens't match the one submitted");
+              event.removeListener();
+              resolve();
+            });
+
+            // After 30s, we throw a timeout error
+            setTimeout(() => {
+              reject(new Error("ERROR: timeout while waiting for confirming hash"));
+            }, 30000);
+          });
+
+          let oracleCheckInterval;
+
+          const rootHashBeforeUpdate = await oracleClient._miner.reputationTree.getRootHash();
+          // Wait for the oracle to follow
+          const oracleUpdated = new Promise(function(resolve, reject) {
+            const checkOracle = async function() {
+              const oracleCurrentHash = await oracleClient._miner.reputationTree.getRootHash();
+              if (oracleCurrentHash !== rootHashBeforeUpdate) {
+                resolve();
+              } else {
+                await forwardTime(1, this);
+              }
+            };
+            oracleCheckInterval = setInterval(checkOracle.bind(this), 1000);
+            setTimeout(() => {
+              reject(new Error("ERROR: timeout while waiting for oracle to update"));
+            }, 100000);
+          });
+
+          // Forward time to the end of the mining cycle and since we are the only miner, check the client confirmed our hash correctly
+          await forwardTime(MINING_CYCLE_DURATION * 0.1, this);
+          await miningCycleComplete;
+
+          await oracleUpdated;
+
+          clearTimeout(oracleCheckInterval);
+          await oracleClient.close();
```

Comment:
Should this end with an `expect` or something to confirm that the oracleClient is synced up with the miner?